extensions/ConfirmEdit/ReCaptcha.class.php

```diff
@@ -7,62 +7,51 @@ class ReCaptcha extends SimpleCaptcha {
 	/**
 	 * Displays the reCAPTCHA widget.
 	 * If $this->recaptcha_error is set, it will display an error in the widget.
-	 *
 	 */
 	function getForm() {
-		global $wgReCaptchaPublicKey, $wgReCaptchaTheme;
-
-		$useHttps = ( isset( $_SERVER['HTTPS'] ) && $_SERVER['HTTPS'] == 'on' );
-		$js = 'var RecaptchaOptions = ' . Xml::encodeJsVar( array( 'theme' => $wgReCaptchaTheme /*Wikia change */ /*, 'tabindex' => 1 */ /*Wikia change end*/) );
+		$siteKey = F::app()->wg->ReCaptchaPublicKey;
+		$theme = SassUtil::isThemeDark() ? 'dark' : 'light';
 
-		return Html::inlineScript( $js ) . recaptcha_get_html( $wgReCaptchaPublicKey, $this->recaptcha_error, $useHttps );
+		return '<div class="g-recaptcha" data-sitekey="' . $siteKey . '" data-theme="' . $theme . '"></div>';
 	}
 
 	/**
 	 * Calls the library function recaptcha_check_answer to verify the users input.
 	 * Sets $this->recaptcha_error if the user is incorrect.
 	 * @return boolean
-	 *
 	 */
 	function passCaptcha() {
-		global $wgReCaptchaPrivateKey, $wgRequest;
+		$wg = F::app()->wg;
 
-		// API is hardwired to return wpCaptchaId and wpCaptchaWord, so use that if the standard two are empty
-		$challenge = $wgRequest->getVal( 'recaptcha_challenge_field', $wgRequest->getVal( 'wpCaptchaId' ) );
-		$response = $wgRequest->getVal( 'recaptcha_response_field', $wgRequest->getVal( 'wpCaptchaWord' ) );
+		$secret = $wg->ReCaptchaPrivateKey;
+		$response = $wg->Request->getText( 'g-recaptcha-response' );
+		$ip = $wg->Request->getIP();
 
-		if ( $response === null ) {
-			// new captcha session
-			return false;
-		}
+		$verifyUrl = 'https://www.google.com/recaptcha/api/siteverify' .
+			'?secret=' . $secret .
+			'&response=' . $response .
+			'&remoteip=' . $ip;
```

Comment:
This can be its own function
```php
protected function getVerifyUrl() {
		$wg = F::app()->wg;
		$secret = $wg->ReCaptchaPrivateKey;
		$response = $wg->Request->getText( 'g-recaptcha-response' );
		$ip = $wg->Request->getIP();

		return 'https://www.google.com/recaptcha/api/siteverify' .
			'?secret=' . $secret .
			'&response=' . $response .
			'&remoteip=' . $ip;
}
```